clang/include/clang/Index/IndexDataStore.h

```diff
@@ -26,9 +27,14 @@ class IndexDataStore {
   ~IndexDataStore();
 
   static std::unique_ptr<IndexDataStore>
-    create(StringRef IndexStorePath, std::string &Error);
+    create(StringRef IndexStorePath,
+      std::map<std::string, std::string, std::greater<std::string>>
+          PrefixMap,
```

Comment:
I agree, and as an input to the API I think conceptually it should be an array of string pairs. Whether then the underlying implementation uses a `std::map` or a `llvm::StringMap` for the lookups, it would be an implementation detail.

Reply:
I was mirroring how Clang stores its prefix maps: https://cs.github.com/llvm/llvm-project/blob/0f304ef0170231b860a249f34e07f50686392253/clang/lib/CodeGen/CGDebugInfo.h#L87 and https://cs.github.com/llvm/llvm-project/blob/0f304ef0170231b860a249f34e07f50686392253/clang/include/clang/Basic/LangOptions.h#L421. I can change this to be more like how Swift does it (they're different, like you mentioned)

Reply:
Ideally those wouldn't be maps either 😅. So these are checked in alphabetical order rather than the order they were given on the CLI?

Reply:
Yes, that's my understanding. I think it would make sense to clean up, but not sure it's in scope for here. Thoughts?

Reply:
I'd say it's worth making whatever you add usable in those cases, but we'd want to fix that upstream anyway (so no, not in scope).